xbmc/pvr/epg/EpgInfoTag.cpp

```diff
@@ -372,7 +374,8 @@ const std::string CPVREpgInfoTag::GetWritersLabel() const
 
 const std::string CPVREpgInfoTag::GetGenresLabel() const
 {
-  return StringUtils::Join(m_genre, CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_videoItemSeparator);
+  return StringUtils::Join(
+      Genre(), CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_videoItemSeparator);
```

Comment:
`return DeTokenize(Genre());` would be more correct here I think, based on your second commit :)

Reply:
You misinterpreted my second commit. `GetGenresLabel` return value is used by GUI engine which expects the genres string separated by whatever `m_videoItemSeparator` is (advanced setting, can be adjusted by the user; default is " / ").

`DeTokenize` uses `EPG_STRING_TOKEN_SEPARATOR` which is a constant (value is ",").

BTW, this PR does not change the logic how genre value is calculated, only when. So, if there is a bug (which I doubt as the respective code is unchanged for a long time), it is not introduced by this PR.

Reply:
No worries. My mistake.